infra/chromadb_compaction/lambdas/enhanced_compaction_handler.py

```
@@ -2360,3 +2358,184 @@ def process_delta_messages(
+    downloaded = 0
+    # Fast path: if bundled tarball exists, download and extract
+    try:
+        tar_key = f"{prefix.rstrip('/')}/delta.tar.gz"
+        s3.head_object(Bucket=bucket, Key=tar_key)
+        local_tar = os.path.join(dest_dir, "delta.tar.gz")
+        os.makedirs(dest_dir, exist_ok=True)
+        s3.download_file(bucket, tar_key, local_tar)
+        import tarfile
+
+        with tarfile.open(local_tar, "r:gz") as tar:
+            tar.extractall(dest_dir)
```

Ruff (0.13.1)

Uses of `tarfile.extractall()`

(S202)